lib/Onyx.js

```diff
@@ -1414,32 +1444,7 @@ function update(data) {
         }
     });
 
-    const promises = [];
-    let clearPromise = Promise.resolve();
-
-    _.each(data, ({onyxMethod, key, value}) => {
-        switch (onyxMethod) {
-            case METHOD.SET:
-                promises.push(() => set(key, value));
-                break;
-            case METHOD.MERGE:
-                promises.push(() => merge(key, value));
-                break;
-            case METHOD.MERGE_COLLECTION:
-                promises.push(() => mergeCollection(key, value));
-                break;
-            case METHOD.MULTI_SET:
-                promises.push(() => multiSet(value));
-                break;
-            case METHOD.CLEAR:
-                clearPromise = clear();
-                break;
-            default:
-                break;
-        }
-    });
-
-    return clearPromise.then(() => Promise.all(_.map(promises, p => p())));
```

Comment:
I see we are undoing the previous logic to move the `clear()` to the front of the update queue. Does running the updates sync solve the issue that caused us to make that change?

Reply:
ah, probably not. I'll take care of it.

Reply:
I'm not sure making sure clear runs first is a desired behavior but I guess now it is too late to change it, other parts of the system might rely on this behavior. I added sorting to the list to make sure clear operations get executed first.

Reply:
Yeah, I agree it is a bit of a weird hack. But changing it could have unexpected consequences. Thanks for updating.